homeassistant/components/ssdp/__init__.py

```diff
@@ -167,6 +169,8 @@ def info_from_entry(entry, device_info):
     }
 
     if device_info:
+        info[ATTR_SSDP_DATA] = device_info
+        # Deprecated below, use from device_info instead
```

Comment:
I think we had the below info for backwards compatibility. But since we can check all the integrations which have an `async_step_ssdp`, we should be able to verify this. Couldn't we just send `device_info?` as the actual thing? :D

Reply:
Since we use `ATTR_` constants, we should be able to update the constants and things keep working ?

Reply:
We can, but for compatibility we then need to inject some things from the SSDP entry to the now UPnP device description data. This feels a bit ugly, but I guess it's bearable. Implemented now.

It would be better in my opinion if those injected values would have clearer constant names and better values; that would not apparently require any in-tree changes at the moment as seemingly nothing uses them yet, like from:
```python
ATTR_HOST = "host"
ATTR_PORT = "port"
ATTR_SSDP_DESCRIPTION = "ssdp_description"
ATTR_ST = "ssdp_st"
```
...to:
```python
ATTR_SSDP_HOST = "ssdp_host"
ATTR_SSDP_PORT = "ssdp_port"
ATTR_SSDP_LOCATION = "ssdp_location"
ATTR_SSDP_ST = "ssdp_st"
```
Actually, I think I'd rather just remove the host and port altogether; things that need them can easily enough urlparse them from the location.

Thoughts?

BTW the hopefully shortly incoming Huawei LTE SSDP support (#28214) does use the SSDP attributes, but I can easily take care of those if the renaming/removal is the way we want to go.

Reply:
Any thoughts about this?

Reply:
Anything I can do to get this moving?

Reply:
Rebased and implemented the above plan.